clang_delta/EmptyStructToInt.cpp

```diff
@@ -213,6 +261,58 @@ void EmptyStructToInt::HandleTranslationUnit(ASTContext &Ctx)
     TransError = TransInternalError;
 }
 
+void EmptyStructToInt::handleOneRecordDecl(const RecordDecl *RD,
+                                           const RecordDecl *BaseRD,
+                                           const FieldDecl *FD,
+                                           unsigned int Idx)
+{
+    // Skip field if it is not ofthe type of one of the structs that are affected by the change
+    // Works recursively through chained structs
+    if(!RecordDeclToField[BaseRD] && BaseRD != TheRecordDecl)
+    {
+        return;
+    }
+
+    IndexVector *NewIdxVec = RecordDeclToField[RD];
+
+    if(!NewIdxVec)
+    {
+        NewIdxVec = new IndexVector();
+        RecordDeclToField[RD] = NewIdxVec;
+    }
+
+    NewIdxVec->push_back(Idx);
+}
+
+void EmptyStructToInt::handleOneVarDecl(const VarDecl *VD)
+{
```

Comment:
The struct we change can still be used as a member type in a different struct. In this case we cannot replace the complete initialiser list and we have to keep tract of the fact that the one struct in used inside the other one. For instance:
```
struct S0 {
   float unused_member;
};

struct S1 {
   struct S1 s;
   int i;
};

struct S1 s1 = {{1.2f}, 3};
```
When getting the callback for the `struct S1 s1` declaration we must know that `struct S1` uses `struct S0` as member type and we can only replace the `{1.2f}` with `0`.